src/search/searchBar.component.tsx

```diff
@@ -12,35 +14,101 @@ import {
   selectSearchParams,
 } from '../state/slices/searchSlice';
 
+export type TimeframeDates = {
+  fromDate: Date | null;
+  toDate: Date | null;
+};
+
 const SearchBar = (): React.ReactElement => {
   const dispatch = useAppDispatch();
 
-  const searchParams = useAppSelector(selectSearchParams);
+  const searchParams = useAppSelector(selectSearchParams); // the parameters sent to the search query itself
   const { dateRange, shotnumRange } = searchParams;
 
-  const [fromDate, setFromDate] = React.useState<Date | null>(
-    dateRange.fromDate ? new Date(dateRange.fromDate) : null
-  );
-  const [toDate, setToDate] = React.useState<Date | null>(
-    dateRange.toDate ? new Date(dateRange.toDate) : null
-  );
+  // ########################
+  // DATE-TIME FIELDS
+  // ########################
+  // The FROM and TO dates sent as part of the query (if any)
+  const [searchParameterFromDate, setSearchParameterFromDate] =
+    React.useState<Date | null>(
+      dateRange.fromDate ? new Date(dateRange.fromDate) : null
+    );
+  const [searchParameterToDate, setSearchParameterToDate] =
+    React.useState<Date | null>(
+      dateRange.toDate ? new Date(dateRange.toDate) : null
+    );
 
-  const [shotnumMin, setShotnumMin] = React.useState<number | undefined>(
-    shotnumRange.min ?? undefined
-  );
-  const [shotnumMax, setShotnumMax] = React.useState<number | undefined>(
-    shotnumRange.max ?? undefined
+  // ########################
+  // TIMEFRAME
+  // ########################
+  // Timeframe range is NOT sent as part of query
+  // Instead, we use this to determine the FROM and TO dates
+  // This is set in the timeframe search box by the user
+  // If the user edits any of the date fields otherwise, this is set to null
+  const [timeframeRange, setTimeframeRange] =
+    React.useState<TimeframeRange | null>(null);
+
+  const calculateTimeframeDateRange = (
+    timeframe: TimeframeRange
+  ): { from: Date; to: Date } => {
+    const to = new Date();
+    const from = new Date();
+
+    switch (timeframe.timescale) {
+      case 'minutes':
+        from.setMinutes(to.getMinutes() - timeframe.value);
+        break;
+      case 'hours':
+        from.setHours(to.getHours() - timeframe.value);
+        break;
+      case 'days':
+        from.setDate(to.getDate() - timeframe.value);
+        break;
```

Comment:
We do have `date-fns` available - `sub` may be useful but since it's relatively simple maths I think we're fine with native functions - if we had to be able to do time frames like "1 day and 1 hour ago" then I think using `sub` would be best, but I don't think we need that.

Reply:
I hadn't thought about using that, I think it's a good shout! Much more reliable

Reply:
It also means we don't need the case statement - we can feed the units directly into `sub`

Reply:
Addressed in https://github.com/ral-facilities/operationsgateway/pull/138/commits/793b2ad3b408be6e047f6895dcb6bb415a4023da